### ISSUE TITLE
nextjs api .. TypeError [ERR_INVALID_URL] 

### ISSUE THREAD

**Issue by evk8** [nrwl/nx#17274]
### Current Behavior

http://localhost:4200/api/hello was not found.
HTTP ERROR 500 occurs.

Please check the issue.

### Expected Behavior

http://localhost:4200/api/hello was not found.
HTTP ERROR 500 occurs.

Please check the issue.

### GitHub Repo

https://github.com/evk8/nx-16.2.2.git

### Steps to Reproduce

1.  nx serve kis

<img width="1686" alt="스크린샷 2023-05-28 오후 7 24 41" src="https://github.com/nrwl/nx/assets/90998467/7f53f458-dcce-46b9-99c6-b89211edde77">

<img width="1345" alt="스크린샷 2023-05-28 오후 7 24 01" src="https://github.com/nrwl/nx/assets/90998467/57f238df-f38d-43da-9d85-3856b123a680">

### Nx Report

```shell
Node   : 16.16.0
   OS     : darwin arm64
   pnpm   : 7.18.0
   Hasher : Native
   
   nx                 : 16.2.2
   @nx/js             : 16.2.2
   @nx/jest           : 16.2.2
   @nx/linter         : 16.2.2
   @nx/workspace      : 16.2.2
   @nx/cypress        : 16.2.2
   @nx/devkit         : 16.2.2
   @nx/eslint-plugin  : 16.2.2
   @nx/next           : 16.2.2
   @nx/react          : 16.2.2
   @nrwl/tao          : 16.2.2
   @nx/web            : 16.2.2
   typescript         : 5.0.4
```

### Operating System

- [X] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

**Comment by ndcunningham** [comment]
I don't think `customServer` works with Next.js new `app/` router

See features here for configuring both:
- [App Router](https://nextjs.org/docs/app/building-your-application/configuring)
- [Pages Router](https://nextjs.org/docs/pages/building-your-application/configuring)

**Comment by evk8** [comment]
Great! Thanks for the review!
Fixed in nextjs 13.4.4. -> https://github.com/vercel/next.js/pull/49805

NX team please put up a default version as well.
